server/neptune/workflows/internal/deploy/revision/queue/worker.go

```diff
@@ -72,30 +89,69 @@ func (w *Worker) Work(ctx workflow.Context) {
-		ctx = workflow.WithValue(ctx, internalContext.DeploymentIDKey, msg.ID)
+		ctx := workflow.WithValue(ctx, internalContext.SHAKey, deployRequest.Revision)
+		ctx = workflow.WithValue(ctx, internalContext.DeploymentIDKey, deployRequest.ID)
 
 		// This should only happen on startup
 		// If we fail to fetch the latest deployment, log and exit the workflow
 		if w.LatestDeployment == nil {
-			w.LatestDeployment, err = w.fetchLatestDeployment(ctx, msg)
+			w.LatestDeployment, err = w.fetchLatestDeployment(ctx, deployRequest)
 			if err != nil {
 				logger.Error(ctx, fmt.Sprint("Unable to fetch latest deployment, worker is shutting down", err.Error()))
 				return
 			}
 		}
 
-		// TODO: Validate revision before executing the workflow
+		// Skip comparing commits if deploy request revision is same as the latest deployed revision
+		if w.LatestDeployment.Revision == deployRequest.Revision {
+			logger.Info(ctx, fmt.Sprintf("Deployed Revision: %s is same as the Deploy Request Revision: %s, moving to next one", w.LatestDeployment.Revision, deployRequest.Revision))
+			w.updateCheckRun(ctx, deployRequest, deployRequest.Repo, IdenticalRevisonSummary)
+			continue
+		}
+
+		var compareCommitResp activities.CompareCommitResponse
+		err = workflow.ExecuteActivity(ctx, w.Activities.CompareCommit, activities.CompareCommitRequest{
+			DeployRequestRevision:  deployRequest.Revision,
+			LatestDeployedRevision: w.LatestDeployment.Revision,
+			Repo:                   deployRequest.Repo,
+		}).Get(ctx, &compareCommitResp)
+		if err != nil {
+			logger.Error(ctx, fmt.Sprintf("Unable to compare deploy request commit with the lates deployed commit: %s", err.Error()))
+		}
+
+		switch compareCommitResp.CommitComparison {
+		case activities.DirectionIdentical:
+			logger.Info(ctx, fmt.Sprintf("Deployed Revision: %s is identical to the Deploy Request Revision: %s, moving to next one", w.LatestDeployment.Revision, deployRequest.Revision))
+			w.updateCheckRun(ctx, deployRequest, deployRequest.Repo, IdenticalRevisonSummary)
+			continue
+
+		case activities.DirectionBehind:
+			logger.Info(ctx, fmt.Sprintf("Deployed Revision: %s is ahead of the Deploy Request Revision: %s, moving to next one", w.LatestDeployment.Revision, deployRequest.Revision))
+			w.updateCheckRun(ctx, deployRequest, deployRequest.Repo, DirectionBehindSummary)
+			continue
+
+		// TODO: Handle Force Applies
+		case activities.DirectionDiverged:
+			logger.Error(ctx, fmt.Sprintf("Deployed Revision: %s is divergent from the Deploy Request Revision: %s.", w.LatestDeployment.Revision, deployRequest.Revision))
+
+		case activities.DirectionAhead:
+			logger.Info(ctx, fmt.Sprintf("Deployed Revision: %s is ahead of the Deploy Request Revision: %s", w.LatestDeployment.Revision, deployRequest.Revision))
+
+		default:
+			logger.Error(ctx, fmt.Sprintf("Invalid commit comparison response: %s", compareCommitResp.CommitComparison))
+			continue
+		}
```

Comment:
Can you move this into a separate function? This makes it hard to grok what's happening in this whole function.